src/main/java/dev/lukebemish/ziggradle/ZigBuildTask.java

```diff
@@ -0,0 +1,104 @@
+package dev.lukebemish.ziggradle;
+
+import org.apache.commons.io.FileUtils;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public abstract class ZigBuildTask extends BaseZigTask<ZigBuildOptions> {
+    @Override
+    @Nested
+    public abstract ZigBuildOptions getOptions();
+
+    @OutputDirectory
+    public abstract DirectoryProperty getWorkingDirectory();
+
+    @OutputDirectory
+    public abstract DirectoryProperty getPrefixDirectory();
+
+    /**
+     * This exists so that gradle can detect changes. Should match the contents of build.zig.zon -> paths
+     */
+    @InputFiles
+    @PathSensitive(PathSensitivity.RELATIVE)
+    public abstract ConfigurableFileCollection getSourceFiles();
```

Comment:
See, this is exactly why having a dedicated task for this seems like a non-goal -- if you're using zig to track builds and whatnot, _also_ using gradle seems like a bad idea.